crates/soft-confirmation-rule-enforcer/src/call.rs

```diff
@@ -35,32 +39,31 @@ impl<C: Context, Da: DaSpec> SoftConfirmationRuleEnforcer<C, Da> {
         address: C::Address,
         context: &C,
         working_set: &mut WorkingSet<C>,
-    ) -> anyhow::Result<CallResponse> {
-        anyhow::ensure!(
-            *context.sender() == self.get_authority(working_set),
-            "Only authority can change the authority"
-        );
-        self.authority.set(&address, working_set);
-        Ok(CallResponse::default())
+    ) -> Result<CallResponse, SoftConfirmationModuleCallError> {
+        if *context.sender() == self.get_authority(working_set) {
+            self.authority.set(&address, working_set);
+            Ok(CallResponse::default())
+        } else {
+            Err(SoftConfirmationModuleCallError::RuleEnforcerUnauthorized)
+        }
     }
 
     pub(crate) fn modify_max_l2_blocks_per_l1(
         &self,
         max_l2_blocks_per_l1: u32,
         context: &C,
         working_set: &mut WorkingSet<C>,
-    ) -> anyhow::Result<CallResponse> {
-        anyhow::ensure!(
-            *context.sender() == self.get_authority(working_set),
-            "Only authority can change the max L2 blocks per L1"
-        );
-
-        let mut data = self.data.get(working_set).expect("Data must be set");
+    ) -> Result<CallResponse, SoftConfirmationModuleCallError> {
+        if *context.sender() == self.get_authority(working_set) {
```

Comment:
```suggestion
        if *context.sender() != self.get_authority(working_set) { return             Err(SoftConfirmationModuleCallError::RuleEnforcerUnauthorized) } 
```
And skip the else clause, makes it clearer imo